linux/task.go

```diff
@@ -72,7 +74,9 @@ func (t *Task) Info() runtime.TaskInfo {
 
 // Start the task
 func (t *Task) Start(ctx context.Context) error {
+	t.mu.Lock()
 	hasCgroup := t.cg != nil
+	t.mu.Unlock()
```

Comment:
Should we defer this one? This will allow the other calls to be successful once the task has started

Reply:
the monitor call would produce a deadlock in that method